packages/react/src/component.tsx

```diff
@@ -6,13 +6,62 @@ import {
   generateMimeTypes,
 } from "uploadthing/client";
 import { useUploadThing } from "./useUploadThing";
-import type { FileRouter } from "uploadthing/server";
+import type { FileRouter, ExpandedRouteConfig } from "uploadthing/server";
 import type { DANGEROUS__uploadFiles } from "uploadthing/client";
 
 type EndpointHelper<TRouter extends void | FileRouter> = void extends TRouter
   ? "YOU FORGOT TO PASS THE GENERIC"
   : keyof TRouter;
 
+const generatePermittedFileTypes = (config?: ExpandedRouteConfig) => {
+  const fileTypes = config ? Object.keys(config) : [];
+
+  const maxFileCount = config
+    ? Object.values(config).map((v) => v.maxFileCount)
+    : [];
+
+  return { fileTypes, multiple: maxFileCount.some((v) => v && v > 1) };
```

Comment:
if an endpoint accepts multiple types, shouldn't it also always allow multiple files?

Reply:
My gut feel is that someone who didn't input limits would expect either an image OR video, and someone customizing limits would want multiple in one upload